docs/architecture/pairing-application-instance-with-compass.md

```diff
@@ -0,0 +1,49 @@
+# Pairing Application Instance with Compass
+
+## Introduction
+
+Application is one of the three basic objects in the Compass. The Application is an abstraction over the actual Application Instance. To fully benefit from the Compass features the Application Instance must be pair with its abstraction. The Compass offers two ways of establishing a trusted relation:
+1. [Pairing without use of Integration System](./pairing-without-using-integration-system.md)
+1. [Pairing with use of Integration System](./pairing-with-use-integration-system.md)
+
+## API specification
+
+`AuthTypeInput` GraphQL input parameter describing the supported authentication type, for example: OAuth2, ClientCertificate.
+
+```graphql
+input AuthTypeInput {
+    type: string!
+}
+```
+
+`PairingAuthResult` GraphQL type that wraps the result returned from the `generateAuthForApplication` mutation. It wraps the authentication details, for example, `client_id` and `client_secret` or a one-time token.
+
+```graphql
+type PairingAuthResult {
+    params: JSON!
+}
+```
+
+`AuthType` GraphQL type that wraps the requested authentication type.
+
+```graphql
+type AuthType {
+    type: string!
+}
+```
+
+The `getAuthTypesForApplication(appId: ID!): [AuthType!]` query returns the possible values for the `type` parameter for auth generation mutation.
+
+```graphql
+type Query {
+    getAuthTypesForApplication(appId: ID!): [AuthType!]
+}
+```
+
+The `generateAuthForApplication(appId: ID!, type: AuthTypeInput!, inputParams: JSON!): PairingAuthResult!` mutation is responsible for providing the authentication details for the Application to perform the pairing.
+
+```graphql
+type Mutation {
+    generateAuthForApplication(appId: ID!, type: AuthTypeInput!, inputParams: JSON!): PairingAuthResult!
+}
+```
```

Comment:
how to register pairing adapter? can we have many paring adapters for one integration system?